### ISSUE TITLE
Feature request: other equal temperaments besides 12edo

### ISSUE THREAD

**Issue by wolftune** [thetwom/Tuner#19]
One of the other ways people tune instruments is to different equal-divisions-of-the-octave or edo https://en.xen.wiki/w/EDO

I'm particularly promoting 41edo as used by Kite-fretting see https://kiteguitar.com/ and the associated up/down ^/v notation standard. That's now supported in some tuners on desktop computers (it's in Guitarix now) and on this simple web tuner which works on desktop or mobile: https://github.com/wilckerson/microtonal-tuner but the native Android app and your wonderful UI would be ideal.

P.S. Love the way this tuner works and that F-Droid is put first and the rest of the promotion of FLO values! Thanks!!

**Comment by thetwom** [comment]
Thanks for the suggestion. The main difficulty to implement non-12-tone tunings is to get integrated nicely into the UI. I would guess, that >99% of the users never heard of such temperaments, so I have to be careful when implementing a feature which makes life harder for them :-). 

Here some quite unordered thoughts which come to my mind when thinking about how to implement it: E.g. would non-12-tone be  a setting for changing temperaments next to 12 tone temperaments? Like having a list EDO12, EDO19, .., Pythagoreon, Just, ...? Or do we need two settings, one for number of notes per octave and one for temperament? What would happen with stored instruments, if they don't fit the temperament? I understand, that 12 tone could be used as subset to 41 edo... so, we could still use these instruments, but other EDOs are not really a subset to 12-tone... so would they have to disappear? Or would we better store the temperament directly with the instrument? Then the tuning editor would need an additional option where one can choose the temperament, but this would make life harder for the 99% of users, since they would not know what this option means. Also coupling temperament with instrument often doesn't make sense since the same instrument could be tuned with different temperaments ... . Or should we just allow edos which have 12-tone as subset and then simple add "minor" ticks to the charts ... so that the user won't be able use the "in-between-notes" as a note of the instrument, but at least see in the pitch plot if it matches his target? 

Quite a lot of text, just showing that I am not sure, how to integrate such a feature in a consistent way without making the UI more complex :-). But would be of course nice to find a good way. 

So, if you have any suggestions or experience, let me know :-). 

That being said, I can't make any promises about if/when I find time to implement such a feature.

**Comment by wolftune** [comment]
Flexible settings are ideal but rare. So, the idea of EDO and then choosing any number and for that matter choosing any arbitrary ratio and dividing it arbitrarily (e.g. the "Bohlen-Pierce" scale is 13 equal divisions of a tripling, meaning an octave+perfect5th instead of dividing the octave). That would be a stand-out remarkable type of thing to support. It can get more nuanced still, tons of variations in possible temperaments.

I think supporting any more-than-12-note temperaments could make it easier to just later add more options to a growing menu.

Storing with an instrument isn't a bad idea, but you're right about some complexities.

As for "make life harder", there's something to be said for presenting temperaments as an option for the huge benefit of introducing the idea. It's similar to software-freedom. Until someone hears about F-Droid and then looks at licenses or about the F-Droid warnings, they might have never thought of the issues, but they should learn somehow. What if Google Play offered some actual respect for software freedom and the ability to filter to freedom-respecting apps? It would be clear that a conscious decision to avoid that in the name of making things easier and simpler would be user-hostile. It's not a perfect analogy. But if when you went to the store, you were offered instruments with various temperaments, you'd have the chance to learn that this is a thing versus today where people think something is wrong with themselves when they can't tune a D chord and a G chord by ear and get both in tune…

So, I'd say temperament should be front-and-center in terms of telling people that it exists, but then people shouldn't be presented with options unless they choose to engage with the topic.

To be clear, 41edo has a lot of notes that get close to 12, but it's not a strict subset, it's not 48edo.

There *can* be 12-note subsets of temperaments, so pick 12 of the commonly-named tones from 41edo and then people can tune to those at least. Having little tick marks for the 41edo steps would be appropriate then.

Thanks for thinking this through, it's certainly not trivial.

**Comment by thetwom** [comment]
Thanks for your further thoughts. I guess we have a bit different views on the situation, but still I think in summary we can agree that supporting non-12-note temperaments would be nice, but it seems not trivial to find a way how integrate it in the UI in a consistently (on the other side, ... if an elegant solution is found it might appear trivial :-). 

So, thinking a bit about it, I guess it will come down to add the non-12-note temperaments to the temperament list (there is already a non-official version supporting 12-tone temperaments) and then try to adapt all functionality, that everything still works the same way as before but with "other" notes. The thing I am most worried about is, that other settings will depend on the temperament, so the user might loose settings without knowing. E.g. the latest non-official version allows to define a frequency for an arbitrary note as reference frequency. So you could also say, my tuning is based on C4 with frequency 300Hz. Switching to another temperament which does not share the same notes would have to change this setting somehow (e.g. reset it to some default) ... and such things should be avoided as much as possible, since the user might not expect, that there are other settings which will be changed. Even worse, e.g. when just for fun switching to 41 EDO and directly switching back to 12 tone he might have lost his reference note without being aware ... 

So, these thoughts don't mean that it is impossible to integrate things nicely, it just means it is not easy without confusing the user. 

I might try to integrate such a thing some time ... but it will definitely be nothing which will be available short-term :-(

**Comment by thetwom** [comment]
sooo ... it took me some time, since the way how notes are handled had to be changed internally ... but this should now all work. 

Here is a first prototype for enabling 41-edo: [v5.0.0-rc1](https://github.com/thetwom/Tuner/releases/tag/v5.0.0-rc1).
Before you install the new version, make sure, that you exported all user-defined instruments. Otherwise, I cannot guarantee, that they will still work and that you can recover them after downgrading! Theoretically, there should be no problems, but it's not tested yet.
It would be great if you could give it a try. You enable it in the settings -> temperaments .. and then it's at the bottom ...

What I won't provide as default, are specific instruments for edo 41 (and other temperaments which are not 12-tone), since they are too exotic to have them as default. But we could discuss, if you provide some typical instruments as user-defined instruments and export it. Then I could place them in an extra folder here on github, such that people could download extra instruments.

**Comment by thetwom** [comment]
It seems I didn't get the symbols fully correct yet ... but before updating this, it would be interesting to hear, which symbols to prefer (e.g. C C^ C^ C#v C# ... OR better C C^ Dbv Db ...)? The second way would avoid double-ups, but  Db would come before C#, which might be a bit confusing ... but since I am not using such kinds of tunings, it is hard to judge.

**Comment by wolftune** [comment]
For 41edo, here's the complete list (the ^ v symbols go in front):

```
D
^D
^^D/vEb
vD#/Eb
D#/^Eb
^D#/vvE
vE
E
^E
vF
F
^F
^^F/vGb
vF#/Gb
F#/^Gb
^F#/vvG
vG
G
^G
^^G/vAb
vG#/Ab
G#/^Ab
^G#/vvA
vA
A
^A
^^A/vBb
vA#/Bb
A#/^Bb
^A#/vvB
vB
B
^B
vC
C
^C
^^C/vDb
vC#/Db
C#/^Db
^C#/vvD
vD
```

So, it's fine enough if it's the right UI choice to choose one of each of the paired examples, preferring whichever seems simpler or some other consistent rule.

I do hope to get to testing the prototype soon. I'm a bit overloaded right now though.

**Comment by wolftune** [comment]
Okay, tested it! Works fine with the initial way you did it. Goes through many symbols, responds well! With the updated symbols, it would work great. Thanks for making testing so easy by providing the simple apk!

**Comment by thetwom** [comment]
Thanks for testing, I am still trying to get the symbols a bit more compact and improve the alignment at some places. So it will take a bit more time for releasing. But it is good to know, that there a no real problems. If you want I can short-term provide a apk with the correct symbols. But as said, I am still trying to improve them a bit before release ...

Do you think other EDOs will be required? Kite guitar only needs edo41, not sure, if others are of interest? I don't want to randomly add more ... better I wait until something is required ...

**Comment by wolftune** [comment]
There are other beyond-12-note temperaments besides EDOs of course, but EDOs are simple and practical. The most popular are 17, 19, 22, 24 (aka quarter-tones), 27, 29, 31, 41, 53… Of those, 31 is the closest competitor to 41 in being most harmonically tuned and not so enormous like 53 and larger. Rarely does any guitarist or acoustic instrument of any sort use more than 31.

Incidentally the Kite-fretting puts in _half_ the frets of 41edo and then tunes the strings so that each pair of strings has half the 41edo pitches. So, the fretting is 20.5edo, exactly half-way between 19edo and 22edo — and those both share some features of Kite layout except they are significantly more out of tune with harmonics. That's the miracle of the Kite guitar — an unexpected mathematical coincidence provides the in-tune harmonies of 41edo (the smallest edo that tunes both fifths and thirds better than 12edo) *and* this weird pattern where all of the harmonies from major-2nd to perfect-fourth show up at every-other-note in the list (which doesn't happen with 31 or 53 or others). So, that's why Kite-fretting is so special, it's not so many frets, playable, and also in-tune better than all smaller edos.

**Comment by thetwom** [comment]
Here an update with several other EDOS and symbol corrections: [v5.0.0-rc2](https://github.com/thetwom/Tuner/releases/tag/v5.0.0-rc2). 
I did modify the symbols, making them more compact, since the original ones were too big and screwed the font alignments ... hope that one can still recognize the arrows ... if you are able to test, let me know your opinion ...

**Comment by wolftune** [comment]
Tested. That seems like a pretty reasonable trade-off choice. There are people who like those symbols with the arrows and the accidentals. Using v and ^ is easier to type casually, but overall, this seems to me to be a good balance. It's certainly functional.

I had a side-thought maybe too complex: delaying the shift to the next note until getting closer to it. So, it currently seems to jump to ^B from B as soon as the pitch is *closer* to ^B. But this means the display shifts suddenly in a relatively unexpected place. What if it got within a few cents of ^B before shifting to it and also within a few cents of B before shifting back to B. So, sliding between the two, it would stick longer at whichever the current one is before shifting over.

**Comment by thetwom** [comment]
Thanks for testing, good that the symbols are working so far. Yes I also considered using v and ^ but this makes the symbols much longer and makes the layouting more difficult. 

About your side-thought. This is already done right now. At the moment the note switch occurs at the ratio 0.6/0.4, meaning that the note switch occurs not exactly at the center between the notes (at the center would be 0.5/0.5). We could consider making the ratio a bit larger, but I don't like to make it too large, since users might start to wonder, why there is no note switch even though they are close to the note. Since it occurred to you, that there is no such "hysteresis effect", I guess the current setting is still a bit close to the center. Maybe, 0.666/0.3333 might be a good compromise. I think, we are ready to release within the next days. Might take some time for me, since it is a bit of work (updating screenshots, ...). After release you still can test and we can talk about adapting the ratio further.

**Comment by wolftune** [comment]
That makes sense, I think the ratio 0.6/0.4 makes sense in 12edo as there is a decent while before someone gets 60% of the way toward the other note. But when the pitches get closer together, the ratio might be better a tad stickier, so yeah, I'd try 0.7/0.3 or 0.75/0.25.

**Comment by thetwom** [comment]
Yes, I had another thought. The distance between two notes in EDO 41 is below 30cents. For EDO 53 almost 20cents. As you were saying, at some point, this is so close that you can hardly tune without accidentally switching between notes. So I introduced now "safety region" of +/- 20 cents, where no switch will occur (well ... there are some conditions, when this is not possible, then the fall back 0.6/0.4 is used ...). I hope, this improves the tuning experience :-). 

Will release as soon as I find time to wrap everything up ...

**Comment by thetwom** [comment]
I filed the release now: [v5.0.0](https://github.com/thetwom/Tuner/releases/tag/v5.0.0). Should appear on play store on fdroid within the next days. Compared to the rcs ,this has several small improvements, also as said this "safety region" for switching notes. If you the app and experience any difficulties or find possible improvements, please let me know. Personally I don't use the non-12-note temperaments, so it is hard for me to judge the actual user experience ...

**Comment by wolftune** [comment]
This could be closed in that it's finished, but I'll add one bit of feedback now that I've used this with 41edo:

Unfortunately, the double-names missing is harder to deal with. It's doable but not ideal. If I'm tuning a note that I know is a v3 (down major third) from another note, it's easy to keep the names in mind (e.g. vF + v3 = vvA ), and sometimes it's harder to think about which enharmonic name is right (in that case ^G♯). There's obviously some cut-off, and we go with avoiding ever using triple-ups or triple-downs.

And the v/^ symbols are easier to see than the little accidental inflection arrows.

All that said, I was able to use the tuner. I'm just saying that I'd prefer the bulkier names, maybe smaller font or something.

Regardless, thanks for your work, it's great to support these more flexible options!

**Comment by thetwom** [comment]
Thanks for you feedback. I really depend on this, since as said I don't use these tunings, so I cannot judge about the actual user experience. 

With "double-names missing", you mean that it would be easier to see the name of both enharmonics and not just one? While I think that I understand the motivation of the ups/down-notation, it appears quite hard to me to have an intuitive way of understanding, what note is higher than another. As long as one needs relative relations between notes it is just fine, I guess.

 So the question is, if there is a way to improve the situation, e.g.
- Showing both emharmonics always -> I don't like this so much, since it displays more information than needed, making it harder to read.
- Use always the enharmonics in such a way, that the "base notes" are ascending in a monotonuous way. Meaning, only to use the left or the right column of the note names you gave before.

Regarding the symbols I totally understand that you are 1. used more to v/^ and 2. that the arrow heads are quite small. However, from the implementation point of view, having these special symbols is more convenient since otherwise the layouting becomes more difficult (we would need to increase margins or decrease the font size dependent on the temperament, must make sure that of he different "plots" stays nice). And personally the v/^ appears to me more like a quick workaround to describe ups and down, since one has nothing better (it's like having to write an arrow as ->, instead of having a real symbol for it ... it works, but it's not as it should be...). But as said, this can be attributed to personal taste.
So, I would prefer trying to improve the symbols for making it more readable. Maybe I can manage to increase the size of the arrowheads enough, that at least it is easier to read.  But I understand, that it still is not the same due to the introduction of the "neutral"-symbol.

What I also would be interested in, is the scenario of tuning these instruments. I would have guessed, that you have a guitar of some type and have to tune always with the same notes. In this scenario, one could simply create a custom instrument such that it is easier associate strings with notes.  But maybe you have to retune depending on the piece of music, so you are better using the chromatic tuning?

**Comment by wolftune** [comment]
> it would be easier to see the name of both enharmonics and not just one?

Yes, in terms of both versions in the cases where I listed doubles in above

>  an intuitive way of understanding, what note is higher than another

Ups and downs are completely logical themselves. Anything with an up is sharper than without it, double-up is sharper than single-up. The less-intuitive part is where ^A♯ is compared to whatever version of B♭. That cannot be intuitive because it *varies* for the different EDOs. The logic is: the *plains* (no ^/v) are the chain of 19 perfect fifths. So, plain B♭ is a perfect 5th down from F which is a 5th down from C. And that will be true in all the tunings, even though they have different sized 5ths. 41edo has nearly pure perfect 5ths, but 17edo does not. So, whether G♯ is higher or lower than A♭ depends on whether the 5ths are larger or smaller (as in 31edo). In 31edo, G♯ is lower than A♭, whereas in 41edo, it's the other way around. See https://en.xen.wiki/w/31edo and https://en.xen.wiki/w/41edo if really curious to learn more

>  is a way to improve the situation

Maybe a simple toggle readily accessible (preferably not via multiple settings menus) for preferring ♯'s vs preferring ♭'s?

>  v/^

Yeah, the neutral symbol does detract from clearly reading. The v/^ symbols would be *ideal* if merely they were stylized so they looked distinct from the letter and carat symbols. They are visually more readable than little full arrows with stems. There's no specific reason that "up" is specifically more correct to be a line with a triangle.

> scenario of tuning these instruments

Although chromatic seems most flexible and ideal, *indeed* I have a Kite guitar (https://kiteguitar.com/) [well, a few actually], and especially if it can be set for 8-strings and so on, there *is* something to specifying target notes for a tuning instead of chromatic.

I often tune by checking fretted notes, and some people might be checking tuning with even movable frets. However, for day-to-day tuning, I *could* choose to just tune open strings, and have 8 specific target pitches instead of chromatic. Yes, there can be some alternate tunings even though there's a standard. So, setting custom instrument tunings could be ideal.

**Comment by thetwom** [comment]
Thanks for the detailed answers. 

I understand that ups and downs are logical, but as you say the mixing with # and b seems not that intuitive. But this is not a problem of the notation system, since it was designed for a specific context and not to be intuitive when describing notes in an absolute system. At least that is how it appears to me :-). 

> Maybe a simple toggle readily accessible (preferably not via multiple settings menus)

I understand that this might be helpful, but we have to keep in mind that the vast majority of users only will use 12-tone temperaments. So, each additional setting on the main screen which is not needed there is making the UI overly complex. So as long as we don't specialize the tuner to be microtuner, it will always be hard to make it ideal. In the end, this means, that we have to find a good compromise, making the high-EDO tuning easier, under the constraint of not making the 12-tone scales more complex. E.g. I could add the enharmonics for EDO41 as you proposed and allow to switch them in the settings. I understand that it is not ideal but at least it would be a compromise. On the other side, I do note like to enharmonics of most EDOS so much, since they are not symmetric (e.g. v# and ^^b, where we have one down for a sharp but two ups for a flat). But in the end the user experience is more important than symmetry ideas ...

> The v/^ symbols would be ideal if merely they were stylized so they looked distinct from the letter and carat symbols. They are visually more readable than little full arrows with stems. There's no specific reason that "up" is specifically more correct to be a line with a triangle.

We could of course redesign the symbols not using "lines with very small triangles" but finding something which looks nice, is more compact and is more easily readable. I am a bit hesitant since I do not really want to invent yet another set of symbols ... . The one I am using now can be found [here](https://en.xen.wiki/w/Alternative_symbols_for_ups_and_downs_notation). The symbols are a bit adapted to be more compact, so  there are some differences, but still it is rather closely related. Sorry, this was a lot of text, but from implementation point of view it would be good to have dedicated symbols for each note "modifier" (as it is now) but I am open to ideas of better symbols even if the would differ from the known symbols.

Thanks again for your feedback. Sorry, that I am making change so difficult, but having all aspects in mind, different users, layouting, implementation, is not easy. At least not for me :-). Still I need input from users to at least know of the weaknesses of this tuner, so thanks a lot for your time and giving all this feedback!

**Comment by wolftune** [comment]
> Sorry, that I am making change so difficult

Your standards are an outlier of high-quality. Some people make projects, have issues open, and then imagine that issue-openers are all entitled demanders and they complain about it. In most cases, people are polite enough but don't engage deeply or respond rapidly, and they maybe get to some things when they do. I would not even entertain the idea that you are making things difficult. It's the complete opposite. You went and took a *complex* difficult request and actively worked on it and are engaging with me. I think you're making this *surprisingly* easy and respectful, and I'm greatly appreciative of how much you've gone way above and beyond.

Anyway, it works out that I like your UI, especially for people using it to practice variable instruments like violin and voice. However, we have other tools such as this web tuner https://microtonal-tuner.herokuapp.com/tuning/41edo and 41edo is now supported in the Guitarix tuner (GNU/Linux only, not Android). So, we're not desperate for your tuner to be perfect. I love that it supports these things now, in part because the inclusion of them in tuners is acknowledgement, validation, and it can introduce people in the world to these existing. Imagine a 12edo musician just glancing at the options and realizing for the first time that 12 isn't all there can be.

I think it's *okay* as is, and it would be a great (but not *necessary*) addition to have the two options of (A) a prefer-sharps / prefer-flats setting *even if hidden deeper in settings* and (B) some non-chromatic way to set target notes for a guitar tuning that supports more than 6 strings and allows choosing which of the enharmonics to use. I don't think either of these are trivial, and I have had all along no *expectation* that this stuff would be implemented, but I'm mentioning the ideas *in case* it's doable enough.

On the symbols, I don't think this tuner is the place to establish the ideal new form of ^/v symbols. That should be led by people engaged in the world of tuning notation.

THANK YOU

**Comment by thetwom** [comment]
Thanks for the further feedback!

Regarding (A), I guess I should at least reconsider the notation a bit to avoid the current confusion we were discussing before. And I could add the enharmonic notes, which are not symmetric ... so this might already help and it could indeed be quite easy to do, since if  I remember correctly, I would not have to change any implementation, but just adapt note definitions.

Regarding (B), I guess you know that you can already define custom instruments with any number of strings. But what you currently can not do, is to save a specific enharmonic for instruments. The note printing is currently a global setting (prefer sharp/flat). But you can at least create a custom instrument, your kite guitar, with the existing notation. You can even save or share the corresponding files ...

Regarding the notation: I still wonder, if it is possible to "combine" the notation systems, like e.g. using double  ^ but not next to each other but stacked and then put it above a #. Yes, this would be a new symbol, but since it would take elements from the widely used ups/downs notation systems, people would maybe be able to recognize it easily. But I really don't know if this works out, one would have to ask more experienced people there. Just trying without knowing if it will work out is quite a high risk of wasting a lot of time :-).

**Comment by wolftune** [comment]
I really appreciate your curiosity and engagement here. Symmetry is an interesting question. The 19-name standard system is obviously not symmetric. Why are E-F and B-C closer than other letters etc. The answer *again* is in the chain of 5ths and the arbitrary history of the naming. (The ♭ symbol started literally as a b, the rounder version to indicate the b that blends with F instead of the b that blends with E). Obviously 41edo isn't going to divide evenly in 2s or 3s. In 41edo, the symmetry of sorts is that all the "imperfect" intervals (2nd, 3rd, 6th, 7th) have 7 versions: vm, m, ^m, mid (~), vM, M, ^M. So, that's single down and ups and a plain for minors and majors plus a mid. The perfect intervals (unison, 4th, 5th, 8ve) don't fit that pattern but have their own sort of logic.

The reason doubles are ever needed is for two reasons: 1, double-up natural is debatably simpler than a down-flat, either way it's two modifiers; 2, the logic of addition, if you start somewhere and do *two* v intervals in a chain, you'll get the letter-counting wrong if you rename it instead of using doubles. We stop at doubles because triples are crazy, and that's the point where flipping to enharmonics makes sense.

Anyway, there is indeed already double-up and double-down accidentals. Musescore includes these:
![double-arrow-accidentals](https://user-images.githubusercontent.com/2801853/187045572-be6aad3b-a042-42c6-b186-438e2857748e.png)

However, we have found in practice, that separating ups and downs from sharps and flats is really a better experience in scores at least. And when you write generically, like saying something is vM6, adding a sharp or natural or flat to that is not helpful. Visually, it helps to see accidentals as going outside of scales and such, and up/down symbols provide tuning indications.

All that said, the combined symbols are out there and they are usable.

**Comment by thetwom** [comment]
actually, I already have created the double-ups/downs symbols since they are needed anyway for edo 53 :-). I quickly tried to adapt the note names and it all seems to work as expected. So at least on this front, we can improve the situation quickly. I only have to figure out the enharmonics for the edos  :-). 

Regarding enharmonics, up now I had the strategy to use the least ups/downs possible, but this goes against, what you are saying that double-ups are simpler than down-flat. So, here again it's where I was lacking experience :-). But as said, I did change the names at least for edo41 now, so it is working as you described, including prefer-flat/sharp. (The change is not published yet .. but it's working locally ...)

> However, we have found in practice, that separating ups and downs from sharps and flats is really a better experience in scores at least. And when you write generically, like saying something is vM6, adding a sharp or natural or flat to that is not helpful. Visually, it helps to see accidentals as going outside of scales and such, and up/down symbols provide tuning indications.

I fully respect what you are saying here and as said, I am relying on your experience, since I can't try by myself. I will keep this in mind. As said before, from implementation and layouting point of view this makes things complicated, so that is why I mainly try to avoid the separated ups/downs. But all that said, when I looked at the symbols on my phone I really did think that the arrowheads are extremely small. But before starting and redesigning a bit, I rather wanted to have it tested first :-).

**Comment by wolftune** [comment]
On the combinations, it's definitely unfortunate to have up-natural instead of just up.

**Comment by thetwom** [comment]
Are you able to test this? [v5.1.0](https://github.com/thetwom/Tuner/releases/tag/v5.1.0-rc1) 
This uses the ups/downs notation with prefixes, but doesn't use v and ^, but more compact symbols. But I think they are similar enough so everyone, who is used to it will understand it. 
Also I refactored the notation of the notes ...

**Comment by wolftune** [comment]
SUPER! Works perfectly, easier to see, not too big, definitely *better* symbols than the ASCII chars v and ^.

By basic logic and other contexts, I could translate the enharmonics as I was tuning my guitar. However, the ideal would simply be a place to tap on the screen to alternate between enharmonics. So, if I see ^^A and I'm thinking "is that the same as B♭?" (when I was expecting B♭), I could tap and swap the naming. Maybe above the instrument choice? I had some other thoughts about how it could work, but that would be all I need I think, because in different cases, I expect different names (even in 12edo), but I don't actually like seeing the messiness of double-names at once or of going through settings to adjust a universal preference. Swapping on the fly would be perfect.

**Comment by thetwom** [comment]
Thanks for the further feedback. Good to hear, that things are improved. 

Maybe it's worth thinking about moving the "prefer sharp/flat" (or maybe it should be called "switch enharmonics") setting from the settings to the toolbar. I am a bit hesitant though, since this would be quite a prominent space which would be occupied. On the other side I understand that note namings for high-EDO-tunings are not trivial, and this would really help, while it would not really hurt the 12-tone-scales. Maybe I should give it a try ...

**Comment by wolftune** [comment]
I think "switch enharmonics" is a good name, and I think 12-edo musicians would like it also, quick access would be good for everyone.

**Comment by thetwom** [comment]
This would look somehow like: [v5.1.0-rc2](https://github.com/thetwom/Tuner/releases/tag/v5.1.0-rc2).
I guess, it does to job, but I am not fully convinced, since it is not fully consistent:
- It puts a "toggle" next to a "go-to somewhere else" icon. So two toolbar icons with completely different behavior.
- The toggle cannot be used (since it is in the background) while defining temperament or reference note.
- Also in the instruments editor, it somehow does not look like it belongs there ...

But as said, apart from that it is doing the job, so I am undecided ...

**Comment by wolftune** [comment]
LGTM!

> two toolbar icons with completely different behavior.

maybe just a slightly different look? Or some padding space between them?

> The toggle cannot be used (since it is in the background) while defining temperament or reference note.

I see, maybe it can be overlaid to also be in that view? Maybe that's too hard. I think it would be nice to have it functioning both in the settings and the instrument editor ideally. 

I like it for where it's usable.

**Comment by wolftune** [comment]
You could do a bizarre for-now workaround: make the button *invisible* so that tapping it still works but it doesn't show in the screens where it doesn't work! HAHA

**Comment by wolftune** [comment]
The main issue (support for beyond-12) is obviously complete. Would you prefer opening separate issues for the outstanding questions about this idea for switching between flats and sharps and the other idea about some way for users to set up custom tunings besides the included ones?

**Comment by thetwom** [comment]
Sorry, for staying a bit quiet here for some time :-). I was not content with the current flat/sharp switch. That is why I did not release these specific changes. I guess, I found a nicer way to place this switch. Obviously, the app lacks a quick view on the very important basic settings: Temperament and reference frequency. At the moment they are "hidden" in the settings menu, but it is quite dangerous, since people quickly can forget, that they changed something, and then their tuning will be completely off ...

So, I am thinking about introducing a bottom bar in portrait mode and a side bar in landscape mode, where I show the most important settings (+ allowing to change them by tapping on it). And in this process I could add the flat/sharp switch as a third item. Lately I was trying to improve the frequency detection further, writing some tests ... so this is currently a bit "delayed". 

Regarding the custom tunings, we seem to have different understandings here. I was talking about custom instruments, and this is already possible. Are you talking about custom temperaments?

**Comment by wolftune** [comment]
I very strongly agree about that risk of frustrating results when people don't realize what tuning reference and temperament they are using.

Yes, I meant custom temperaments.

**Comment by thetwom** [comment]
So, finally here is a version with a bottom bar and quick settings: [v5.2.0-rc1](https://github.com/thetwom/Tuner/releases/tag/v5.2.0-rc1).

Regarding custom temperaments, this is a difficult request. It's not about allowing to use them, but more the UI for setting them up including note names. For now I have no plans to implement this. I am suggesting, that you open a new issue for it, describing in a few words why this would help. So at least the request becomes visible to others and we can better sort out what is exactly needed.

**Comment by wolftune** [comment]
Opened separate issue about custom temperaments.

That approach of temperament and tuning reference and #/b control at the bottom is great, works wonderfully! I have no feedback or complaints!

**Comment by thetwom** [comment]
This is good to hear. I really like this more now. Of course we cannot switch now inside the reference note and temperament dialogs, but with the current layout it just does not feel like it should be possible :-). 

I will wait for the official release before closing. So this stays visible, in case people are wondering ... (it's more a principle, I don't think that people are really looking in here that they start wondering ;-)
